spec/requests/user_subscriptions_spec.rb

```diff
@@ -0,0 +1,174 @@
+      get subscribed_user_subscriptions_path, params: valid_params
+
+      expect(response).to have_http_status(:ok)
+      expect(JSON.parse(response.body)["is_subscribed"]).to eq true
+    end
+
+    it "returns false if a user is not already subscribed" do
+      valid_params = { source_type: "Article", source_id: 999 }
+      get subscribed_user_subscriptions_path, params: valid_params
+
+      expect(response).to have_http_status(:ok)
+      expect(JSON.parse(response.body)["is_subscribed"]).to eq false
+    end
+  end
+
+  describe "POST /user_subscriptions - UserSubscriptions#create" do
+    it "creates a UserSubscription" do
+      article = create(:article, user: super_admin_user, body_markdown: "---\ntitle: User Subscription#{rand(1000)}\npublished: true\n---\n\n{% user_subscription 'CTA text' %}")
+      valid_attributes = { source_type: article.class_name, source_id: article.id, subscriber_email: user.email }
+      expect do
+        post user_subscriptions_path,
+             headers: { "Content-Type" => "application/json" },
+             params: { user_subscription: valid_attributes }.to_json
+      end.to change(UserSubscription, :count).by(1)
+
+      user_subscription = UserSubscription.last
+      expect(user_subscription.subscriber_id).to eq user.id
+      expect(user_subscription.author_id).to eq article.user_id
+      expect(user_subscription.user_subscription_sourceable_type).to eq article.class_name
+      expect(user_subscription.user_subscription_sourceable_id).to eq article.id
+    end
+
+    it "returns an error for an invalid source_type" do
+      invalid_source_type_attributes = { source_type: "NonExistentSourceType", source_id: "1", subscriber_email: user.email }
+      expect do
+        post user_subscriptions_path,
+             headers: { "Content-Type" => "application/json" },
+             params: { user_subscription: invalid_source_type_attributes }.to_json
+      end.to change(UserSubscription, :count).by(0)
+
+      expect(response).to have_http_status(:unprocessable_entity)
+      expect(response.parsed_body["error"]).to eq("invalid source_type")
+    end
+
+    it "returns an error for a source that can't be found" do
+      invalid_source_attributes = { source_type: "Article", source_id: "99999999" }
+      expect do
+        post user_subscriptions_path,
+             headers: { "Content-Type" => "application/json" },
+             params: { user_subscription: invalid_source_attributes }.to_json
+      end.to change(UserSubscription, :count).by(0)
+
+      expect(response).to have_http_status(:unprocessable_entity)
+      expect(response.parsed_body["error"]).to eq("source not found")
+    end
+
+    it "returns an error for an inactive source" do
+      unpublished_article = create(:article, user: super_admin_user, body_markdown: "---\ntitle: User Subscription#{rand(1000)}\npublished: false\n---\n\n{% user_subscription 'CTA text' %}")
```

Comment:
Not blocking at all, but at some point we might want to clean these specs up a bit for readability and extract this into a factory or trait